test-runner-frontend/src/Views/Report.js

```diff
@@ -0,0 +1,390 @@
+import CloseIcon from '@mui/icons-material/Close';
+import { makeStyles, Dialog } from '@material-ui/core';
+import { Stack } from '@mui/system';
+import MoreHorizIcon from '@mui/icons-material/MoreHoriz';
+import { buildChartsData, buildJsonResults, downloadJSON, printPDF } from './ReportUtils';
+
+const GRAPHQL_JAVA_GITHUB = 'https://github.com/graphql-java/graphql-java';
+
+const useStyles = makeStyles({
+  dialog: {
+    position: 'absolute',
+    right: 0,
+    top: 0,
+    width: 1000,
+    height: 1000,
+    marginBottom: 0,
+  },
+});
+
+function TransitionSnackBar(props) {
+  return (
+    <Slide
+      {...props}
+      direction="up"
+      style={{ color: 'white', backgroundColor: '#e535ab', fontWeight: '600', textShadow: '1px 3px black' }}
+    />
+  );
+}
+
+const initialState = {
+  // State of the dialog that shows test results with JSON format.
+  openDialog: false,
+  // Data that will be used to build charts.
+  classesAndBenchmarksState: [],
+  // Test results with JSON format.
+  jsonResult: {},
+  // Copy of the selected test run from Dashboard
+  selectedTestRunFromDashboardCopy: {},
+  // State of the snackbar that appears after clicking on downloading and copy buttons.
+  openSnackBar: false,
+  // Whether we are loading data.
+  loadingState: true,
+  // Snackbar message according to type of button clicked.
+  snackBarMessage: '',
+  // Snackbar duration according to type of button clicked.
+  snackBarMessageDuration: null,
+  // Snackbar transition state.
+  transitionSnackBar: undefined,
+};
+
+const reducer = (state, action) => {
+  switch (action.type) {
+    case 'handleDialog':
+      return {
+        ...state,
+        openDialog: action.payload.isOpen,
+        jsonResult: action.payload.jsonResults,
+      };
+    case 'setClassesAndBenchmarksState':
+      return {
+        ...state,
+        classesAndBenchmarksState: action.payload,
+        loadingState: false,
+      };
+    case 'setSelectedTestRunFromDashboardCopy':
+      return {
+        ...state,
+        selectedTestRunFromDashboardCopy: action.payload,
+      };
+    case 'handleSnackBar':
+      return {
+        ...state,
+        transitionSnackBar: action.payload.transition,
+        snackBarMessage: action.payload.message,
+        snackBarMessageDuration: action.payload.messageDuration,
+        openSnackBar: action.payload.isOpen,
+      };
+    default:
+      return state;
+  }
+};
+
+export default function Report() {
+  const [state, dispatch] = useReducer(reducer, initialState);
+
+  const {
+    openDialog,
+    classesAndBenchmarksState,
+    jsonResult,
+    selectedTestRunFromDashboardCopy,
+    openSnackBar,
+    loadingState,
+    snackBarMessage,
+    snackBarMessageDuration,
+    transitionSnackBar,
+  } = state;
+
+  const location = useLocation();
+  const classes = useStyles();
+  const { selectedTestRunFromDashboard } = location.state ? location.state : selectedTestRunFromDashboardCopy;
```

Comment:
hmm im a little confused why we have two potential objects that have the data we care about. we should aim to only have one otherwise it gets a little confusing as to what is the source of truth

Reply:
Done.